bpf/lib/utils.h

```diff
@@ -88,18 +88,20 @@ static inline __u32 bpf_ktime_get_sec(void)
 	return (__u64)(bpf_ktime_get_nsec() / NSEC_PER_SEC);
 }
 
-#if __BYTE_ORDER == __LITTLE_ENDIAN
+#if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
+	__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
I don't know the strict rules on whose job it is to define these depending on your compiler, but given that we only ever use LLVM for this code and this is pretty similar to the existing code, I'm assuming this is correct and works.

Reply:
We explicitly need the byte order define from LLVM here. See also: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/lib/bpf/bpf_endian.h

Reply:
These are defined by both GCC (https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html) and LLVM and it's e.g. what the Abseil C++ library uses: https://github.com/abseil/abseil-cpp/blob/c44657f55692eddf5504156645d1f4ec7b3acabd/absl/base/config.h#L444-L465